boards/arm/mm_swiftio/burner/pyocd_user.py

```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2020, MADMACHINE LIMITED
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+
+def will_connect():
+    # Look up the external flash memory region.
+    extFlash = target.memory_map.get_region_by_name("flexspi")
+
+    # Set the path to an .FLM flash algorithm.
+    flm_path = os.getenv('ZEPHYR_BASE') + '/boards/arm/mm_swiftio/burner/MIMXRT105x_QuadSPI_4KB_SEC.FLM'
```

Comment:
Could just use `pathlib.Path(__file__).parent / 'MIMXRT105x_QuadSPI_4KB_SEC.FLM'` to keep this self-contained.

Reply:
Thanks so much. I fixed it.  I'm a beginner in python and don't know how to deal with this problem, now you provide the answer to me。 I'll close https://lists.zephyrproject.org/g/devel/topic/add_pyocd_user_scripy_cause/71021588?p=,,,20,0,0,0::recentpostdate%2Fsticky,,,20,2,0,71021588

Reply:
using pathlib is a good idea, I test __file__ in other file is OK, below is test code.
```
import pathlib
import platform

print(platform.python_version())
print(pathlib.Path(__file__).parent)
```
But the user script in pyocd can't access global __file__, below is modify code
```
import pathlib

def will_connect():
    # Look up the external flash memory region.
    global target
    extFlash = target.memory_map.get_region_by_name("flexspi")

    # Set the path to an .FLM flash algorithm.
    flm_path = pathlib.Path(__file__).parent / 'MIMXRT105x_QuadSPI_4KB_SEC.FLM'
    extFlash.flm = flm_path
```
When I do west flash will report can't find __file__
```
0000612:CRITICAL:__main__:uncaught exception: name '__file__' is not defined
Traceback (most recent call last):
  File "/home/frank/.local/lib/python3.6/site-packages/pyocd/__main__.py", line 343, in run
    self._COMMANDS[self._args.cmd](self)
  File "/home/frank/.local/lib/python3.6/site-packages/pyocd/__main__.py", line 473, in do_flash
    with session:
  File "/home/frank/.local/lib/python3.6/site-packages/pyocd/core/session.py", line 287, in __enter__
    self.open()
  File "/home/frank/.local/lib/python3.6/site-packages/pyocd/core/session.py", line 367, in open
    self._board.init()
  File "/home/frank/.local/lib/python3.6/site-packages/pyocd/board/board.py", line 80, in init
    self.delegate.will_connect(board=self)
  File "/home/frank/.local/lib/python3.6/site-packages/pyocd/core/session.py", line 412, in __call__
    self._fn(**args)
  File "<string>", line 13, in will_connect
NameError: name '__file__' is not defined
```
Do you have any other suggestions for these? Thanks

Reply:
That's very strange. Please feel free to disregard this comment, then. Thanks.